samples: matter: update manifest and disable Read Client (#13464)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst

.. note::
  The :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option is set to ``y`` by default.
  To disable removing application-specific non-volatile data when the :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT` Kconfig option is selected, set the :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option to ``n``.

> Read Client feature

Reviewer: Super minor: How about `configuration`? `Feature` word seems a little bit off to me in this context. Using server or client is more like selecting a device role or configuration in Interaction Model. When I see feature I expect some bigger functionality that enables something extra besides basic protocol functionality. But yeah, maybe it's only me feeling such difference.

Reviewer 2: We could use just "functionality".

=========================

> Read Client functionality needs to be enabled if the device needs to read attributes from another device in the Matter network.

Reviewer 3: What about: The **Read Client** feature shall be activated when the device requires the capability to access attributes from a different device within the Matter network.

> Read Client functionality needs to be enabled if the device needs to read attributes from another device in the Matter network.

Reviewer 2: Adapting Łukasz's suggestion a bit:
```suggestion
The Read Client functionality is needed for a device to be able to read attributes from another device in the Matter network.
```

> To enable it set the :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` to ``y`` and to ``n`` to disable.

Reviewer 2: ```suggestion
```
The Kconfig is mentioned again two lines later, so this one can go.

> In Matter samples we have disabled this functionality by default for all samples except :ref:`matter_light_switch_sample`, and :ref:`matter_bridge_app` because both of them need to read attributes from the bound devices.

Reviewer 2: ```suggestion
This functionality is disabled by default for Matter samples in the |NCS|, except for ones that need to read attributes from the bound devices, such as the :ref:`matter_light_switch_sample` sample and the :ref:`matter_bridge_app` application.
```
Agree that having an inclusive list is not a good idea, so making the current ones into examples solves the need to update this line unless the functionality would be disabled by default for either of the mentioned ones.

> In Matter samples we have disabled this functionality by default for all samples except :ref:`matter_light_switch_sample`, and :ref:`matter_bridge_app` because both of them need to read attributes from the bound devices.

Reviewer 3: It will be hard to track it with time e.g. which examples have or not have this capability.

I would remove 150 and 151 (which is dup of 149).

Maybe just say that this config is disabled by default.

> The feature shall be activated when the device requires the capability to access attributes from a different device within the Matter network, for example, using bindings.
> This functionality is disabled by default for Matter samples in the |NCS|, except for ones that need to read attributes from the bound devices, such as the :ref:`matter_light_switch_sample` sample and the :ref:`matter_bridge_app` application.

Reviewer 2: ```suggestion
This functionality is disabled by default for Matter samples in the |NCS|, except for ones that need to read attributes from the bound devices, such as the :ref:`matter_light_switch_sample` sample and the :ref:`matter_bridge_app` application.
Enable the feature if your device needs to be able to access attributes from a different device within the Matter network to, for example, use bindings.
```
Not 100% sure if it should be "to, for example, use binding" as I suggested, or "using, for example, bindings".

Author: It should be "using, for example, bindings" :)

> If your device needs to have the Read Client functionality enabled set the :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` to ``y``.

Reviewer 2: ```suggestion
To enable the Read Client functionality, set the :kconfig:option:`CONFIG_CHIP_ENABLE_READ_CLIENT` Kconfig option to ``y``, or to ``n`` to disable it.
```
